cli/commands/run/server/agent.controller.js

```diff
@@ -55,7 +60,12 @@ module.exports = class AgentController {
         const txEvent = this.createTransactionEventFromGrpcRequest(
           call.request
         );
-        findings.push(...(await this.handleTransaction(txEvent)));
+
+        const returnedFindings = await this.handleTransaction(txEvent);
+
+        if(returnedFindings.length > 10) throw Error("Found more than 10 findings when executing transaction handler.")
```

Comment:
lets refactor these checks into a utils method so we're keeping the code DRY. the message can always be the same (i.e. dont need to specify whether it happened from transaction/block handler)